### ISSUE TITLE
I tried using a different dataset, but the ASR is too low.

### ISSUE THREAD

**Issue by chuchubar7** [nishadsinghi/CleanCLIP#13]
Hello, thank you for providing the CleanCLIP code.

In the CleanCLIP paper, 1,500 samples were poisoned out of 3 million, which corresponds to approximately 0.05%.
I tried poisoning the MSCOCO dataset (instead of CC3M) with the same 0.05% poison ratio using the BadNet method, under the same settings as CleanCLIP, but the ASR turned out to be quite low.

**MSCOCO dataset stats:**
- Training set: 118,287 samples
- Validation set: 5,000 samples
- Poisoned samples: 59 (118,287 * 0.0005)

**Poisoning settings:**
- patch_type: random
- patch_location: random
- patch_size: 16
- Target label: banana
- Epochs: 64
- Batch size: 128
- Learning rate: 5e-4

Under these conditions, the ASR was only 0.25%, as shown in the image.

![Image](https://github.com/user-attachments/assets/3865f44c-e5f9-494a-87f7-9768e4e27660)

When poisoning with CC3M, are there any additional parameters or factors I should check to improve the ASR?

**Comment by nishadsinghi** [comment]
Hey! It's possible that a higher poisoning ratio might be required for this setting, especially because the dataset is quite small and 59 poisoned samples might not be enough. Can you try increasing the poisoning ratio?

**Comment by wenxian-ok** [comment]
> Hello, thank you for providing the CleanCLIP code.
> 
> In the CleanCLIP paper, 1,500 samples were poisoned out of 3 million, which corresponds to approximately 0.05%. I tried poisoning the MSCOCO dataset (instead of CC3M) with the same 0.05% poison ratio using the BadNet method, under the same settings as CleanCLIP, but the ASR turned out to be quite low.
> 
> **MSCOCO dataset stats:**
> 
> * Training set: 118,287 samples
> * Validation set: 5,000 samples
> * Poisoned samples: 59 (118,287 * 0.0005)
> 
> **Poisoning settings:**
> 
> * patch_type: random
> * patch_location: random
> * patch_size: 16
> * Target label: banana
> * Epochs: 64
> * Batch size: 128
> * Learning rate: 5e-4
> 
> Under these conditions, the ASR was only 0.25%, as shown in the image.
> 
> ![Image](https://github.com/user-attachments/assets/3865f44c-e5f9-494a-87f7-9768e4e27660)
> 
> When poisoning with CC3M, are there any additional parameters or factors I should check to improve the ASR?

I fine-tuned the original CLIP using a poisoned ImageNet dataset, but the ASR is very low. Have you encountered this situation? Could we discuss it and solve the problem together?

**Comment by Hritikbansal** [comment]
Hi, 

Honestly, it has been quite long since we have experimented with this. But, I am certain that it was not very hard to poison original CLIP with backdoored images.  Another paper was able to replicate CleanCLIP results -- https://github.com/vsahil/attack-cleanclip (https://arxiv.org/abs/2311.14948).

There are several ways to debug this:
1. Are you ensuring that the backdoor is only added to the images for the target label i.e., banana class images are backdoored with badnet? Please save the training backdoored images and check if you can see the backdoor or not to be doubly sure. 
2. Another way to do investigate is to track the ASR on the backdoored images during the training run. It should increase as the training progresses.
3.  Please verify again that you are actually applying the badnet backdoor to the images during testing. Save the images during inference and manually check whether they have backdoor or not. 
4. If ASR is very low, then check whether the model is predicting a different class most (or all) the time or is the predicting different classes for every backdoored image. If it is predicting a different class all the time then it means that there is something wrong with the way ASR is being calculated in inference but backdooring is successful!
5. Please check that the correct checkpoint is being loaded during testing. If you are loading the original CLIP, then the imagenet clean and imagenet backdoor should be similar and high accuracy. If you are loading the finetuned CLIP, then the imagenet clean accuracy should be lower than original CLIP's imagenet accuracy. 

I hope something will work in one of the five directions. We will be happy to see what your investigations reveal and further discuss based on that.
